common/src/main/java/com/wynntils/wynn/model/quests/QuestManager.java

```diff
@@ -0,0 +1,190 @@
+    }
+
+    public static void toggleTracking(QuestInfo questInfo) {
+        CONTAINER_QUERIES.toggleTracking(questInfo);
+    }
+
+    public static void openQuestOnWiki(QuestInfo questInfo) {
+        final String baseUrl = "https://wynncraft.fandom.com/wiki/";
+
+        // TODO handle mini quest
+        String name = questInfo.getName();
+        String wikiQuestPageNameQuery = WebManager.getApiUrl("WikiQuestQuery");
+        String url = wikiQuestPageNameQuery + WebUtils.encodeForCargoQuery(name);
+        Request req = new RequestBuilder(url, "WikiQuestQuery")
+                .handleJsonArray(jsonOutput -> {
+                    String pageTitle = jsonOutput
+                            .get(0)
+                            .getAsJsonObject()
+                            .get("_pageTitle")
+                            .getAsString();
+                    Utils.openUrl(baseUrl + WebUtils.encodeForWikiTitle(pageTitle));
+                    return true;
+                })
+                .build();
+
+        RequestHandler handler = new RequestHandler();
+
+        handler.addAndDispatch(req, true);
+    }
+
+    public static QuestInfo getCurrentQuest() {
+        return currentQuest;
+    }
+
+    protected static void setCurrentQuest(QuestInfo questInfo) {
+        currentQuest = questInfo;
+        WynntilsMod.postEvent(new TrackedQuestUpdateEvent(currentQuest));
+    }
+
+    protected static void setQuests(List<QuestInfo> newQuests) {
+        quests = newQuests;
+        WynntilsMod.postEvent(new QuestBookReloadedEvent.QuestsReloaded());
+    }
+
+    protected static void setMiniQuests(List<QuestInfo> newMiniQuests) {
+        miniQuests = newMiniQuests;
+        WynntilsMod.postEvent(new QuestBookReloadedEvent.MiniQuestsReloaded());
+    }
+
+    protected static void setDialogueHistory(List<List<String>> newDialogueHistory) {
+        dialogueHistory = newDialogueHistory;
+        WynntilsMod.postEvent(new QuestBookReloadedEvent.DialogueHistoryReloaded());
+    }
+
+    /** Shared between the container query classes */
+    protected static String getQuestBookTitle(int pageNum) {
+        return "^§0\\[Pg. " + pageNum + "\\] §8.*§0 Quests$";
+    }
+
+    private static class LocationComparator implements Comparator<QuestInfo> {
+        private static final Vec3 PLAYER_LOCATION = McUtils.player().position();
+
+        private double getDistance(Optional<Location> loc) {
+            // Quests with no location always counts as closest
+            if (loc.isEmpty()) return 0f;
+
+            Location location = loc.get();
+            return PLAYER_LOCATION.distanceToSqr(location.toVec3());
+        }
+
+        @Override
+        public int compare(QuestInfo quest1, QuestInfo quest2) {
+            Optional<Location> loc1 = quest1.getNextLocation();
+            Optional<Location> loc2 = quest2.getNextLocation();
+            return (int) (getDistance(loc1) - getDistance(loc2));
+        }
+    }
```

Comment:
This can reasonably be extracted to another class

Reply:
What do you mean? It is another class already. Or do you mean that it should be a top-level class in its own file? If so, I disagree. This is just a small helper class; it was slightly too large to be a lambda, but it's only used in sortQuestInfoList.

Reply:
I feel that it might be used in other places, but we can move it if that occurs. Indeed, it is only used here for now.